bzt/utils.py

```diff
@@ -627,6 +632,24 @@ def install(self):
             except BaseException as exc:
                 raise exc
 
+    def install_with_mirrors(self, dest, suffix):
+        self.log.info("Will try to install %s into %s", self.tool_name, dest)
+        downloader = request.FancyURLopener()
+        tool_dist = tempfile.NamedTemporaryFile(suffix=suffix, delete=False)  # delete=False because of Windows
+        mirrors = self.mirror_manager.mirrors()
+        for mirror in mirrors:
+            with ProgressBarContext() as pbar:
+                try:
+                    socket.setdefaulttimeout(5)
+                    self.log.debug("Downloading: %s", mirror)
+                    downloader.retrieve(mirror, tool_dist.name, pbar.download_callback)
+                    return tool_dist
+                except BaseException:
+                    self.log.error("Error while downloading %s", mirror)
+                    continue
+                finally:
+                    socket.setdefaulttimeout(None)
```

Comment:
Of course, but not sure if engine should set it.
Should we provide global timeout option in configuration and set it in engine module?